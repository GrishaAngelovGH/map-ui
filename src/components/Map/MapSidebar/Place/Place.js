const Place = ({ title, open }) => (
  <div className='row mb-3 border border-3 rounded shadow p-3 bg-light small'>
    <div className='col-md-10'>
      <p className='fw-bold text-center'>{title}</p>
      <p className={`p-0 alert alert-${open ? 'success' : 'danger'} text-center`}>{open ? 'Open' : 'Closed'}</p>
      <p className='m-0 text-secondary'>4.3 <i className='bi bi-star-fill text-warning'></i> (3037 votes)</p>
      <p className='m-0 text-secondary'>Contemporary all-day European dining</p>
      <p className='m-0 text-secondary'>{open ? 'Closes' : 'Opens'} 11 PM</p>
    </div>
    <div className='col-md-2'>
      <i className='bi bi-check-square-fill text-primary fs-1'></i>
    </div>
  </div>
)

export default Place