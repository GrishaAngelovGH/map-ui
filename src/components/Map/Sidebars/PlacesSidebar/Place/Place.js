const Place = ({ id, title, open, onClick }) => {
  const handleClick = () => {
    onClick(id)
  }

  return (
    <div className='row mb-3 border border-3 rounded shadow p-3 bg-light small'>
      <div className='col-md-10'>
        <p className='fw-bold text-center'>{title}</p>
        <p className={`p-0 alert alert-${open ? 'success' : 'danger'} text-center`}>{open ? 'Open' : 'Closed'}</p>
        <p className='m-0 text-secondary'>4.3 <i className='bi bi-star-fill text-warning'></i> (3037 votes)</p>
        <p className='m-0 text-secondary'>Contemporary all-day European dining</p>
        <p className='m-0 text-secondary'>{open ? 'Closes' : 'Opens'} 11 PM</p>
      </div>
      <div className='col-md-2'>
        <i role='button' className='bi bi-chevron-right bg-primary text-white fs-1 rounded' onClick={handleClick}></i>
      </div>
    </div>
  )
}

export default Place