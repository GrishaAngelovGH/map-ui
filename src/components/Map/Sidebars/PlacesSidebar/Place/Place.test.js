import { render } from '@testing-library/react'

import Place from './Place'

test('should render open Place component', () => {
  const view = render(<Place title='Place' open={true} />)

  expect(view).toMatchSnapshot()
})

test('should render closed Place component', () => {
  const view = render(<Place title='Place' open={false} />)

  expect(view).toMatchSnapshot()
})