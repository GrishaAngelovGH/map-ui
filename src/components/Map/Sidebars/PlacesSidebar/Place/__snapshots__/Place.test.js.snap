// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render closed Place component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
      >
        <div
          class="col-md-10"
        >
          <p
            class="fw-bold text-center"
          >
            Place
          </p>
          <p
            class="p-0 alert alert-danger text-center"
          >
            Closed
          </p>
          <p
            class="m-0 text-secondary"
          >
            4.3 
            <i
              class="bi bi-star-fill text-warning"
            />
             (3037 votes)
          </p>
          <p
            class="m-0 text-secondary"
          >
            Contemporary all-day European dining
          </p>
          <p
            class="m-0 text-secondary"
          >
            Opens
             11 PM
          </p>
        </div>
        <div
          class="col-md-2"
        >
          <i
            class="bi bi-chevron-right bg-primary text-white fs-1 rounded"
            role="button"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
    >
      <div
        class="col-md-10"
      >
        <p
          class="fw-bold text-center"
        >
          Place
        </p>
        <p
          class="p-0 alert alert-danger text-center"
        >
          Closed
        </p>
        <p
          class="m-0 text-secondary"
        >
          4.3 
          <i
            class="bi bi-star-fill text-warning"
          />
           (3037 votes)
        </p>
        <p
          class="m-0 text-secondary"
        >
          Contemporary all-day European dining
        </p>
        <p
          class="m-0 text-secondary"
        >
          Opens
           11 PM
        </p>
      </div>
      <div
        class="col-md-2"
      >
        <i
          class="bi bi-chevron-right bg-primary text-white fs-1 rounded"
          role="button"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render open Place component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
      >
        <div
          class="col-md-10"
        >
          <p
            class="fw-bold text-center"
          >
            Place
          </p>
          <p
            class="p-0 alert alert-success text-center"
          >
            Open
          </p>
          <p
            class="m-0 text-secondary"
          >
            4.3 
            <i
              class="bi bi-star-fill text-warning"
            />
             (3037 votes)
          </p>
          <p
            class="m-0 text-secondary"
          >
            Contemporary all-day European dining
          </p>
          <p
            class="m-0 text-secondary"
          >
            Closes
             11 PM
          </p>
        </div>
        <div
          class="col-md-2"
        >
          <i
            class="bi bi-chevron-right bg-primary text-white fs-1 rounded"
            role="button"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
    >
      <div
        class="col-md-10"
      >
        <p
          class="fw-bold text-center"
        >
          Place
        </p>
        <p
          class="p-0 alert alert-success text-center"
        >
          Open
        </p>
        <p
          class="m-0 text-secondary"
        >
          4.3 
          <i
            class="bi bi-star-fill text-warning"
          />
           (3037 votes)
        </p>
        <p
          class="m-0 text-secondary"
        >
          Contemporary all-day European dining
        </p>
        <p
          class="m-0 text-secondary"
        >
          Closes
           11 PM
        </p>
      </div>
      <div
        class="col-md-2"
      >
        <i
          class="bi bi-chevron-right bg-primary text-white fs-1 rounded"
          role="button"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
